docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+Note it also calls two(2) other utility functions called...
+
+#### Function: `self.get_token_translation`
+
+```python
+
+# NOTE TO READER: we will not have a model until the next "MODE", but mentally note the code explanation here
+
+def get_token_translation(self, translationVal):
+  model = tf.keras.saving.load_model(f'{self.provider.tool_config["modelConfig"]["modelName"]}-exported', custom_objects={'custom_standardization': custom_standardization})
+
+  try:
+    vect_layer = model.layers[0]
+    translation_tokens = [token.strip(" ") for token in translationVal.split(",")]
+    vect_vocab = vect_layer.get_vocabulary()
+    translations = [{"token": int(token), "translation": vect_vocab[int(token)]} for token in translation_tokens]
+  
+    self.info(str(translations))
+    return translations
+  except Exception as e:
+    logger.error(f"get_token_translation: Error occured \n {repr(e)}")
+    self.provider.io.error("An error occured during token translation :( . Please see PythonSdk log for details")
+```
+
+`model = tf.keras.saving.load_model(f'{self.provider.tool_config["modelConfig"]["modelName"]}-exported', custom_objects={'custom_standardization': custom_standardization})`
+
+Here we load a model that(soon) will contain a `TextVectorization` TensorFlow layer at `model.layers[0]` where `translational` is a comma-separated value of integer token values to be "translated" to their original string representations as mapped by said `model.layers[0]` layer.
+
+
+For now, we won't have any `translationVal`s, so this won't be relevant until later sections.
+Then, we log in to `self.info` as defined below:
+
+#### Function: `self.get_token_translation`
+
+```python
+def info(self, s: str):
+  """Log s, and then send an `info` message to designer containg s"""
+  logger.info(s)
+  self.provider.io.info(s)
+```
+
+While the method is relatively straightforward, it includes(spellcheck) a notable point: there is a distinct difference between **logging** and **messaging**.
+We chose this for many reasons, a large one being that the tool _User_ rarely needs, and even less so *wants*, to see the debugging output.
+Do not use the above as a crunch or without consideration; logging is cheap, but _messaging_ is expensive!
+A good rule of thumb is to keep your `self.provider.io` calls to a minimum, only sending what your USER needs to see.
+You can leverage `logging` more freely, especially when paired with the available debugging flag (NOTE: reference needed here).
+Use sparingly, or you will risk slowing your tool with expensive IO! _(A very important consideration in ML)_
+
+Finally, update `on_complete` to call our `send_preview_data()` function:
+
+```python
+self.provider = provider
+# ...
+self.info("Plugin initialized.")
+if self.MODE == 'DATA':
+  # ...
+elif self.MODE == 'PREVIEW':
+  self.send_preview_data()
+```
+
+Now, that we have data and a way to feed it as input into our model we are ready to move on to the next mode.
```

Comment:
TW review:

Now, that we have data and a way to feed it as input into the model, we are ready to move on to the next mode.